### ISSUE TITLE
Track Task field changes

### ISSUE THREAD

**Issue by VassilIordanov** [NCI-Agency/anet#1574]
Provide the ability to track the history of a designated field (customFieldEnum1) of a task.

User should be prompted to add a note when changing the field value
User should be able to see the history of previous note/value/author changes

Needed by end of April

Note: there is a workaround field assessment field in https://github.com/NCI-Agency/anet/pull/1494
This is to be used temporarily until the generic notes are in place. We can convert then these values to notes upon upgrade

**Comment by VassilIordanov** [comment]
ooops, i made a duplicate of #1524

**Comment by maradragan** [comment]
From #1524 :-1: 
From @VassilIordanov :
The concept of task assessments/notes entered every time a status of a task is changed. We discussed this one before a bit. I think a good option may be a mechanism that uses notes to:
· Force that a note is entered when some event happens – i.e. changing a field
· Attach a history event to the note – showing that a field changed from state 1 to state 2. I am not sure if the history event should have a life on its own or be a special kind of note, but that’s probably an implementation detail.

This sounds suspiciously like report approvals/comments/etc. How would we do it if we had to do it again?
